File: L1_insertion_sort.py
import random
from array import array

N = 10
list = array('i', [0] * N)

def insertion_sort(A, n):
    # Traverse the list or array from index 1 to n-1.
    for i in range(1, n):
        key = A[i]

        # Insert A[i] into the sorted subarray A[0:i].
        # Compare stored key with the already sorted values to its left.
        # Move each item one position to the right until we find the
        # position for the key or fall off the left end of the list or array.
        j = i - 1
        while j >= 0 and A[j] > key:
            A[j + 1] = A[j]
            j -= 1

        # Insert key at the correct position in the list or array.
        A[j + 1] = key

def print_list(A, n):
    for i in range (0, n):
        print(" %3d" % (A[i]), end="")
    print()

def main():
    print("LAB 1 - INSERTION SORT")

    for i in range(0, N):
        list[i] = random.randint(0, 99)

    print("LIST BEFORE SORT")
    print_list(list, N)

    print("LIST AFTER SORT")
    insertion_sort(list, N)
    print_list(list, N)


if __name__ == "__main__":
	main()

File: L1_insertion_sort.c
#include <stdio.h>
#include <stdlib.h>

#define N 10

int array[N];

void print_list(int a[], int n) {
    for (int i = 0; i < n; i++) {
        printf("%3d ", a[i]);
    }
    printf("\n");
}

void InsertionSort(int a[], int n) {
    for (int i = 1; i < n; i++) {
        int key = a[i];
        int j = i - 1;

        while (j >= 0 && a[j] > key) {
            a[j + 1] = a[j];
            j -= 1;
        }
        a[j + 1] = key;
    }
}

int main(void) {

    printf("Lab 1 - insertion sort\n");
    for(int i = 0; i < N; i++){
        array[i] = rand() >> 24;
    }

    printf("List before sorting\n");
    print_list(array, N);

    InsertionSort(array, N);

    printf("List after sorting\n");
    print_list(array, N);
    return 0;
}
